### PR TITLE
Refactor: Moving StorageClusterInit in a new file.

### PR DESCRIPTION
**Pull request by rajatsingh25aug**
Moved the logic for `StorageClusterInit` in a new file for better clarity and understanding.

Signed-off-by: RAJAT SINGH <rajasing@redhat.com>

### CODE REVIEW

**Comment by rajatsingh25aug** [comment]
> 1. CSV changes should be a separate commit. In fact, they aren't even needed.

Ok I will  make them in a separate commit. But why they aren't  needed?.
> 2. Changes to `pkg/apis/ocs/v1/zz_generated.deepcopy.go` is not relevant here. Remove it.

These changes came from running `make update generated` and `make latest-gen-csv`
> 3. Changes to `purgeOSDScript` in `pkg/controller/storagecluster/reconcile.go` should not be a part of this PR.
> 4. `shellcheck-stable/README.txt` is not relevant. Remove it.

File: pkg/controller/storagecluster/reconcile.go [hunk 1]
```diff
@@ -1253,13 +1203,13 @@ func newCleanupJob(sc *ocsv1.StorageCluster) *batchv1.Job {
 	const purgeOSDScript = `
 set -x
 
-osd_status=$(ceph osd tree | grep "osd.${FAILED_OSD_ID} " | awk '{print $5}') 
-if [[ "$osd_status" == "up" ]]; then 
+osd_status=$(ceph osd tree | grep "osd.${FAILED_OSD_ID} " | awk '{print $5}')
+if [[ "$osd_status" == "up" ]]; then
```
Comment (anubhav-here) [inline pkg/controller/storagecluster/reconcile.go:1207]: They look like unnecessary changes

File: pkg/controller/storagecluster/reconcile.go [hunk 2]
```diff
@@ -1253,13 +1203,13 @@ func newCleanupJob(sc *ocsv1.StorageCluster) *batchv1.Job {
...
   echo "OSD ${FAILED_OSD_ID} is up and running."
   echo "Please check if you entered correct ID of failed osd!"
-else 
+else
   echo "OSD ${FAILED_OSD_ID} is down. Proceeding to mark out and purge"
-  ceph osd out osd.${FAILED_OSD_ID} 
+  ceph osd out osd.${FAILED_OSD_ID}
```
Comment (anubhav-here) [inline pkg/controller/storagecluster/reconcile.go:1212]: Even here, these changes are not required!

File: pkg/controller/storagecluster/reconcile.go (hunk 1 above)
Comment (rajatsingh25aug) [reply pkg/controller/storagecluster/reconcile.go:1207]: I think these changes came from extra spaces. Fixing it.

Comment (rajatsingh25aug) [reply pkg/controller/storagecluster/reconcile.go:850]: This is correct IMO as the previous contributor **added extra spaces**, my vscode settings remove unnecessary extra whitespace. I didn't add/removed any spaces by myself.

File: pkg/controller/storagecluster/reconcile.go (hunk 2 above)
Comment (rajatsingh25aug) [reply pkg/controller/storagecluster/reconcile.go:855]: This is correct IMO as the previous contributor **added extra spaces**, my vscode settings remove unnecessary extra whitespace. I didn't add/removed any spaces by myself.

File: pkg/controller/storagecluster/reconcile.go (hunk 1 above)
Comment (anubhav-here) [reply pkg/controller/storagecluster/reconcile.go:850]: Woah! Then thats good :)

**Comment by rajatsingh25aug** [comment]
> Code looks good. Please rename the file to `storageclusterinit.go` without camel case.

Done, please merge.

**Comment by obnoxxx** [comment]
@jarrpa @umangachapagain @rajatsingh25aug  -  this should have been reviewed a bit more carefully....

The first patch accidentally removes a hunk of code from reconcile.go that it did not mean to remove (part of the purge-osd-script).
Then the second patch re-introduces that hunk but messes up the whitespacing. Amusingly, this second patch is called "Removed extra whitespaces added by previous contributor". The whitespaces that this patch effectively removes were actually not extra but  good and intentional (indentation of blocks).

The fact that this has not been caught in reviews now makes backporting of #589 unnecessarily difficult.

I wish a bit more scrutiny to be applied to the commits and commit messages in reviewing PRs. :wink:

**Comment by rajatsingh25aug** [comment]
> @jarrpa @umangachapagain @rajatsingh25aug - this should have been reviewed a bit more carefully....
> 
> The first patch accidentally removes a hunk of code from reconcile.go that it did not mean to remove (part of the purge-osd-script).
> Then the second patch re-introduces that hunk but messes up the whitespacing. Amusingly, this second patch is called "Removed extra whitespaces added by previous contributor". The whitespaces that this patch effectively removes were actually not extra but good and intentional (indentation of blocks).

But may I know why do we need extra spaces at the end of a line?. My vscode removes any unnecessary extra whitespaces from everywhere(end of a line, lots of extra space between lines).

> The fact that this has not been caught in reviews now makes backporting of #589 unnecessarily difficult.
> 
> I wish a bit more scrutiny to be applied to the commits and commit messages in reviewing PRs. 😉

**Comment by obnoxxx** [comment]
And now working on re-adding the indentation, I see what must have been @rajatsingh25aug 's intention:
To remove the *trailing* white spaces. That would have been fine (in a separate PR).
In addition, and probably unintentionally, the indentation was removed too.

First of all this should not have been part of this PR.

Secondly, it was only spotted because the accidentally removed block of code. And re-adding was done with sacrificing white spaces at the beginning of the line, and that was put as a separate commit that claimed only to remove white spaces, but in fact it added back a whole block of code. :-)

**Comment by rajatsingh25aug** [comment]
I see now. It should've been a separate PR. My bad that I didn't thought of
this earlier.
But now, what should we do next?. Should I raise a new PR?.
Anything else that you can suggest?.

**Comment by obnoxxx** [comment]
@rajatsingh25aug writes:
> But may I know why do we need extra spaces at the end of a line?

Ah, at the end of the line, we don't usually need them.
But such unrelated changes should go in a separate PR.

> My vscode removes any unnecessary extra whitespaces from everywhere(end of a line, lots of extra space between lines).

Apparently, your editor tried to be a bit too clever here....
Because this was part of a string that actually contains a script, it may not have understood that this is a script and removed the indentation.

In general, I am not in favour of tools that interfere too much with the editing.
At the very least we should always critically check if the tool has done the right thing.

The first accident in this PR was not the whitespace change though, but the fact that first patch https://github.com/openshift/ocs-operator/commit/a79a94584b8969038e349fde79674f15efbca75c removed almost the whole script. And the second commit https://github.com/openshift/ocs-operator/commit/74c06f2b51c49640f3ec6c3a88236bb01172d2a7 brought back the script but with whitespace changes.

So the intentions were good. But two problems:
1. error in the first commit undone in the second commit - this is what makes backports difficult.
2. the editor removed too many white spaces.

Cheers - Michael

**Comment by obnoxxx** [comment]
@rajatsingh25aug writes:
> I see now. It should've been a separate PR. My bad that I didn't thought of this earlier.

No issues, this is also what reviews are for. Review should have been more diligent. :wink: 
@anubhav-here actually spotted that there was something strange, but convinced all is good.

> But now, what should we do next?. Should I raise a new PR?.

I have it almost ready. 

> Anything else that you can suggest?.

No, keep this feed-back in mind. Otherwise, keep up the good work. :smiley:

**Comment by rajatsingh25aug** [comment]
thanks a lot Miachel 😬 and again from next time, we will keep these nifty
things in mind.
